src/main/java/de/codingair/codingapi/player/data/PacketReader.java

```diff
@@ -104,23 +104,16 @@ public void unInject() {
     public synchronized void unInject(@Nullable Runnable later) {
         JavaPlugin plugin = API.getInstance().getMainPlugin();
 
-        Runnable runnable = () -> {
-            if (currentHandler != null && channel != null && player.isOnline()) {
-                try {
-                    channel.pipeline().remove(this.currentHandler);
-                } catch (Throwable ignored) {
-                    //thrown when this handler is not registered anymore -> ignore
-                }
+        if (currentHandler != null && channel != null && player.isOnline()) {
+            try {
+                channel.pipeline().remove(this.currentHandler);
+            } catch (Throwable ignored) {
+                //thrown when this handler is not registered anymore -> ignore
             }
+        }
```

Comment:
Please do this in an executor as mentioned in your comment that pointed to [HolographicDisplays](https://github.com/filoghost/HolographicDisplays/blob/70779348af6870d711a5ad322ffe47aec981397b/nms/v1_16_r3/src/main/java/me/filoghost/holographicdisplays/nms/v1_16_R3/VersionNMSManager.java#L99).